### PR TITLE
Add rule for Transmission

### PR DESCRIPTION
**Pull request by tlm-2501**
I have absolutely no idea what I'm doing. Please confirm this looks right before I go about creating more of these.

### CODE REVIEW

File: jiup/rules/rules.go [hunk 1]
```diff
@@ -1476,6 +1476,18 @@ func init() {
...
+	Rule("transmission",
+	     v.HTML(
+		     "https://transmissionbt.com/download/",
+		     "#current-version",
+		     "innerText",
+	     ),
+	     d.HTMLA(
+		     "https://transmissionbt.com/download/",
+		     "#msi32-link>a",
+		     "#msi64-link>a"
```
Comment (geek1011) [inline jiup/rules/rules.go]:
```suggestion
		     "#msi64-link>a",
```

File: jiup/rules/rules.go [hunk 2]
```diff
@@ -1476,6 +1476,18 @@ func init() {
 			return x86, x64, nil
 		},
 	)
+	Rule("transmission",
+	     v.HTML(
+		     "https://transmissionbt.com/download/",
+		     "#current-version",
+		     "innerText",
+	     ),
```
Comment (geek1011) [inline jiup/rules/rules.go]:
```suggestion
		     nil,
	     ),
```

**Comment by geek1011** [comment]
You've got the right idea. You can have a look at https://godoc.org/github.com/just-install/just-install-updater-go/jiup/rules for the syntax of the functions. Thanks for working on this!

**Comment by tlm-2501** [comment]
I've tried looking at the source code and I'm having trouble figuring things out still, but at least I got what parameters functions accept and what the mysterious `d` and `v` things are. Is this PR fine for merge now, or is there more left to do?

**Comment by geek1011** [comment]
It would be good if the site didn't use JS to fill in the values. You'll need to use the regexp extractor to extract the version from https://transmissionbt.com/includes/js/constants.js, and the template download to generate the msi link (`https://github.com/transmission/transmission-releases/raw/master/transmission-{{.Version}}-` `x86` or `x64` `.msi`).

d and v are the download/version extractors. The docs are at the bottom of the godoc page.

**Comment by tlm-2501** [comment]
Yyyyiiiikes. I did not realise JS is used for this. Hold on, I'm going to figure this out.

**Comment by tlm-2501** [comment]
I *think* it should be fine now, but of course I'm still not completely sure.

**Comment by geek1011** [comment]
Yep. It's fine. Inkscape just changed and is making it fail (I'm fixing that now). Thanks!

**Comment by geek1011** [comment]
FYI, here's the updated Inkscape rule: https://github.com/just-install/just-install-updater-go/commit/46a166845b05a65b7336fa1794ff4898930b15ed?diff=unified&w=1

**Pull request merged**